Toolbox/arcsdm/areafrequency.py

```diff
@@ -39,369 +38,257 @@
+    tblrow2 = tblrows2.next()
+    tblrow2 = tblrows2.next()
     while tblrow2:
         Eff_CumAre.append(tblrow2.Eff_CumAre)
         Cum_Sites.append(tblrow2.Cum_Sites)
-        tblrow2 = tblrows2.Next()
-
-    # Finally, calculate the Eff_AUC field from two lists and Efficiency
+        tblrow2 = tblrows2.next()
+    
     sumEff_AUC = 0.0
-    tblrows1 = gp.UpdateCursor(Output_Table)
-    tblrow1 = tblrows1.Next()
+    tblrows1 = arcpy.UpdateCursor(Output_Table)
+    tblrow1 = tblrows1.next()
     for i in range(len(Eff_CumAre)):
         val = 0.5 * (tblrow1.Eff_CumAre - Eff_CumAre[i]) * (tblrow1.Cum_Sites + Cum_Sites[i]) / (100.0 * 100.0)
         sumEff_AUC += val
-        # gp.AddMessage(str(val))
         tblrow1.Eff_AUC = val
-        tblrows1.UpdateRow(tblrow1)
-        tblrow1 = tblrows1.Next()
-    # Calculate last row
+        tblrows1.updateRow(tblrow1)
+        tblrow1 = tblrows1.next()
+    
     if tblrow1:
-        # gp.AddMessage("Calculating last row...")
         val = 0.5 * (tblrow1.Eff_CumAre) * (tblrow1.Cum_Sites) / (100.0 * 100.0)
         sumEff_AUC += val
         tblrow1.Eff_AUC = val
-        # gp.AddMessage(str(val))
-        tblrows1.UpdateRow(tblrow1)
-    del tblrow1,tblrows1
-
-    gp.addmessage('Efficiency: %.1f%%'%(sumEff_AUC*100.0))
-
-    if Input_table and joinRastername: # In case of joined integer raster and table
-        gp.RemoveJoin_management(joinRastername, Input_table)
-            
-            
-    # except UserException:
-    #     print('User exception caught. ')
-        
-    # except arcpy.ExecuteError:
-    #     #TODO: Clean up all these execute errors in final version
-    #     gp.AddMessage("AreaFrequency caught: arcpy.ExecuteError");
-    #     gp.AddMessage("-------------- END EXECUTION ---------------");        
-    #     raise 
-    # except:
-    #     #In case of joined integer raster and table
-    #     arcpy.AddMessage("Tsip");
-    #     if Input_table and joinRastername:
-    #         gp.RemoveJoin_management(joinRastername, Input_table)
-    #    # get the traceback object
-    #     tb = sys.exc_info()[2]
-    #     # tbinfo contains the line number that the code failed on and the code from that line
-    #     tbinfo = traceback.format_tb(tb)[0]
-    #     # concatenate information together concerning the error into a message string
-    #     pymsg = "PYTHON ERRORS:\nTraceback Info:\n" + tbinfo + "\nError Info:\n    " + \
-    #         str(sys.exc_type)+ ": " + str(sys.exc_value) + "\n"
-    #     # generate a message string for any geoprocessing tool errors
-    #     msgs = "GP ERRORS:\n" + gp.GetMessages(2) + "\n"
-    #     gp.AddError(msgs)
-
-    #     # return gp messages for use with a script tool
-    #     gp.AddError(pymsg)
-
-    #     # print messages for use in Python/PythonWin
-    #     print (pymsg)
-    #     print (msgs)
-    #     raise
+        tblrows1.updateRow(tblrow1)
+    
+    del tblrow1, tblrows1
```

Comment:
I would replace this with:
```
    with arcpy.da.SearchCursor(Output_Table, ["Eff_CumAre", "Cum_Sites"]) as cursor:
        for row in cursor:
            Eff_CumAre.append(row[0])
            Cum_Sites.append(row[1])
    
    sumEff_AUC = 0.0
    with arcpy.da.UpdateCursor(Output_Table, ["Eff_CumAre", "Cum_Sites", "Eff_AUC"]) as cursor:
        for i, row in enumerate(cursor):
            if i < len(Eff_CumAre):
                val = 0.5 * (row[0] - Eff_CumAre[i]) * (row[1] + Cum_Sites[i]) / (100.0 * 100.0)
                sumEff_AUC += val
                row[2] = val
                cursor.updateRow(row)
            else:
                val = 0.5 * row[0] * row[1] / (100.0 * 100.0)
                sumEff_AUC += val
                row[2] = val
                cursor.updateRow(row)
```

Reply:
I have edited the code and commited new code.